src/main/java/com/minecolonies/coremod/permissions/ColonyPermissionEventHandler.java

```diff
@@ -218,31 +219,40 @@ public void on(final PlayerInteractEvent event)
 
             if(Configurations.enableColonyProtection)
             {
-                if(!perms.hasPermission(event.getEntityPlayer(), Permissions.Action.RIGHTCLICK_BLOCK) && event.getWorld().getBlockState(event.getPos()).getBlock() != null)
+                boolean cancel = false;
+
+                if((event.getWorld().getBlockState(event.getPos()) != null) &&
+                        !(event.getItemStack().getItem() instanceof ItemFood) &&
+                        !perms.hasPermission(event.getEntityPlayer(), Permissions.Action.RIGHTCLICK_BLOCK))
                 {
-                    cancelEvent(event, event.getEntityPlayer());
+                    cancel = true;
                 }
 
-                if(event.getWorld().getBlockState(event.getPos()).getBlock() instanceof BlockContainer && !perms.hasPermission(event.getEntityPlayer(),
-                        Permissions.Action.OPEN_CONTAINER))
+                if(event.getWorld().getBlockState(event.getPos()).getBlock() instanceof BlockContainer &&
+                        !perms.hasPermission(event.getEntityPlayer(), Permissions.Action.OPEN_CONTAINER))
                 {
-                    cancelEvent(event, event.getEntityPlayer());
+                    cancel = true;
                 }
 
-                if(event.getWorld().getTileEntity(event.getPos()) != null && !perms.hasPermission(event.getEntityPlayer(), Permissions.Action.RIGHTCLICK_ENTITY))
+                if(event.getWorld().getTileEntity(event.getPos()) != null &&
+                        !perms.hasPermission(event.getEntityPlayer(), Permissions.Action.RIGHTCLICK_ENTITY))
                 {
-                    cancelEvent(event, event.getEntityPlayer());
+                    cancel = true;
                 }
 
-                if(event.getItemStack() != null && event.getItemStack().getItem() instanceof ItemPotion && !perms.hasPermission(event.getEntityPlayer(),
-                        Permissions.Action.THROW_POTION))
+                if(event.getItemStack() != null && event.getItemStack().getItem() instanceof ItemPotion &&
+                        !perms.hasPermission(event.getEntityPlayer(), Permissions.Action.THROW_POTION))
                 {
-                    cancelEvent(event, event.getEntityPlayer());
+                    cancel = true;
                 }
 
                 if(event.getItemStack() != null && event.getItemStack().getItem() instanceof ItemScanTool
                         && !perms.hasPermission(event.getEntityPlayer(), Permissions.Action.USE_SCAN_TOOL))
                 {
+                    cancel = true;
+                }
+
+                if (cancel) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Move this left curly brace to the beginning of next line of code. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3ALeftCurlyBraceStartLineCheck)